content/blog/2019-12-18-qvec/index.md

```diff
@@ -0,0 +1,145 @@
+
+Quantum computing offers unique computing properties due to the nature of a qubit. The main computational differences between quantum and classical computing include the following properties:
+
+|  Property       |  Quantum    | Classical CPU |
+| ------  | ------------- | ------------------- |
+| Architecture   | Spatial | Von-Neumann  |
+| Data    | Quantum State | Voltage Level |
+| Control | External (Laser) | Voltage Level |
+| States per bit | Exponential | Linear |
+
+In general, quantum computer can implement all of the computation primitives that a classical one can. Both, in theory, can be turing complete with a universal set of logic gates. However, quantum computing also has computational primitives that classical computers don't share. These primitives are key to quantum supremacy: the concept that a quantum computer can theoretically outperform classical computers.
+
+| Unique compute | Example Usage |
+| ----------------| ------------- |
+| Large State Space  | Chemical Reaction Simulation |
+| Entanglement | Combinatorial Optimization (TSP) |
+| Amplitude Magnification | Database Search |
+| Probabilistic (multiple results) | ? |
+| Phase | ? |
+
+A potential downside to quantum computing is that it is inherently probabilistic. An output on one execution may be entirely different from the output on the next run. Quantum algorithms must be designed so that the correct answer must have measurement probability >50%. The answer can then be inferred by repeating the execution many times and taking the majority result. Many quantum algorithms exist in Bounded Quantum polynomial class (BQP) where the correct answer can be found in polynomial time with probability at least 2/3.  It can, however, practically be time and resource intensive to run quantum programs a sufficient number of times to achieve a reasonable confidence.
+
+## Opportunities for Vectorization
+
+The probabilistic nature requires that multiple repeated runs of the same program be executed. The number of runs required to obtain a "correct" result depends on one's error threshold and the design of the algorithm; specifically, the number of runs to obtain error `e` is given as `O(log(1/e))`. Thus, there are diminishing returns for running the algorithm many times, but it is important to run the algorithm a "reasonable" amount to achieve acceptably low error.
+
+The naive method to repeatedly apply the algorithm is to run many iterations of the algorithm sequentially. This serialization can potentially increase the runtime depending on how many repeated applications are required. Consider, for instance, the entanglement program below, which must be run a large number of times to produce a correct result:
```

Comment:
Maybe briefly mention what language these examples will be written in (Scaffold)?